pkgs/build-support/writable-dir-wrapper/default.nix

```diff
@@ -0,0 +1,190 @@
+{
+  lib,
+  writeShellApplication,
+  xorg,
+}:
+/**
+  A wrapper for programs that require a writable directory to function, typically
+  games or proprietary software that expect read-only resources to be placed alongside
+  config files and state files (e.g. logs).
+
+  # Inputs
+
+  *`options`* (Attribute set)
+
+  : Set of options for the wrapper.
+
+    `name` (String)
```

Comment:
Is there any reason to actually change the name of the script here?

I think it would be better to just have a static name similar to `make{,Binary}Wrapper` rather than each derivation possibly having their own name for this script. It'd probably avoid a lot of somewhat confusing install/fixupPhases